src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -218,15 +233,133 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
+		boolean lotUpdated = false;
+		boolean lotAdded = false;
+		boolean medicalAdded = false;
+
+		// TODO: To decide if to make allMedicals parameter
+		boolean allMedicals = true;
+		List<Movement> movs = new ArrayList<Movement>();
+		List<Medical> inventoryMedicalsList = inventoryRowSearchList.stream()
+						.map(MedicalInventoryRow::getMedical)
+						.distinct()
+						.collect(Collectors.toList());
+		if (allMedicals) {
+			// Fetch all movements without filtering by medical code
+			movs.addAll(movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null));
+		} else {
+			// Fetch only movements concerning inventoryRowSearchList list
+			for (Medical medical : inventoryMedicalsList) {
+				movs.addAll(movBrowserManager.getMovements(medical.getCode(), null, null, null, movFrom, movTo, null, null, null, null));
+			}
+		}
+
+		// Cycle fetched movements to see if they impact inventoryRowSearchList
+		for (Movement mov : movs) {
+			Lot movLot = mov.getLot();
+			String lotCodeOfMovement = movLot.getCode();
+			String lotExpiringDate = TimeTools.formatDateTime(movLot.getDueDate(), TimeTools.DD_MM_YYYY);
+			String lotInfo = GeneralData.AUTOMATICLOT_IN ? lotExpiringDate : lotCodeOfMovement;
+			Medical medical = mov.getMedical();
+			String medicalDesc = medical.getDescription();
+			Integer medicalCode = medical.getCode();
+
+			// Fetch also empty lots because some movements may have discharged them completely
+			Optional<Lot> lot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCodeOfMovement))
+							.findFirst();
+			double mainStoreQty = lot.get().getMainStoreQuantity();
+
+			// Search for the specific Lot and Medical in inventoryRowSearchList (Lot should be enough)
+			Optional<MedicalInventoryRow> matchingRow = inventoryRowSearchList.stream()
+							.filter(row -> row.getLot().getCode().equals(lotCodeOfMovement) && row.getMedical().getCode().equals(medicalCode))
+							.findFirst();
+
+			if (matchingRow.isPresent()) {
+				MedicalInventoryRow medicalInventoryRow = matchingRow.get();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				if (mainStoreQty != theoQty) {
+					// Update Lot
+					medicalInventoryRow.setTheoreticQty(mainStoreQty);
+					MedicalInventoryRow updatedRow = medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+					if (updatedRow != null) {
+						lotUpdated = true;
+						medDescriptionForLotUpdated
+										.append("\n")
+										.append(MessageBundle.formatMessage(
+														"angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.detail.fmt.msg",
+														medicalDesc, lotInfo, theoQty, mainStoreQty, mainStoreQty - theoQty));
+					}
+				}
+			} else {
+				MedicalInventoryRow newMedicalInventoryRow = new MedicalInventoryRow(null, mainStoreQty, mainStoreQty, inventory, medical,
+								mov.getLot());
```

Comment:
Please keep it separated anyway, for more code readability:
```suggestion
				// TODO: to decide if to give control to the user about this
				double realQty = mainStoreQty;
				MedicalInventoryRow newMedicalInventoryRow = new MedicalInventoryRow(null, mainStoreQty, realQty, inventory, medical,
								mov.getLot());
```